### DISCUSSION TITLE
improper image header error - DDS conversion

### DISCUSSION THREAD

**Discussion by wazix11** [ImageMagick/ImageMagick#4062] [Help]
I am trying to batch convert DDS files to PNG files, some of the files convert just fine, but about 2/3 of them give this error:

mogrify: improper image header `Apex_Rock_grey_C.dds' @ error/dds.c/ReadDDSImage/1932.

They all give the same error (aside from the file name).

All of these files can be opened with programs like Paint.NET and Gimp and converted to PNG, but I would like to do this with commands as there are 1,500 files left to convert.

Any help would be appreciated.

**Comment by dlemstra** [comment]
Please tell us the exact version of ImageMagick that you are using and is it possible to share an example file so we can explain better why this is happening?

**Reply by wazix11** [reply]
I am using ImageMagick-7.1.0-4-Q16-HDRI-x64

[Apex_Rock_grey_C.zip](https://github.com/ImageMagick/ImageMagick/files/6957828/Apex_Rock_grey_C.zip)

**Reply by dlemstra** [reply]
The trick is to zip the file: [Apex_Rock_grey_C.zip](https://github.com/ImageMagick/ImageMagick/files/6955333/Apex_Rock_grey_C.zip)

**Reply by wazix11** [reply]
I tried reinstalling the newest version (Version: ImageMagick 7.1.0-4 Q16 x64 2021-07-18) as well as switching commands from "magick mogrify -format png *.dds" to "magick convert -format png *.dds" as well as "magick *.dds png" all with the same error aside from the "mogrify" part switching to either "convert" or "magick"

**Reply by fmw42** [reply]
You cannot have wild cards in the input and output. Also use magick, not magick convert. Try one DDS file for input and whatever you want for output. Please always show full commands.

**Reply by wazix11** [reply]
Using command: magick Apex_Rock_grey_C.dds Apex_Rock_grey_C.png
returns error: magick: improper image header `Apex_Rock_grey_C.dds' @ error/dds.c/ReadDDSImage/1932.

**Reply by fmw42** [reply]
Apparently your DDS file is written improperly. What software or camera created your file? Did you do any other processing on the file before uploading it? I get the same message on IM 7.1.0.4 Q16 Mac OSX Sierra.

**Reply by wazix11** [reply]
These are images extracted from a game I play, the majority of the files convert properly but about 2/3 of the texture map files are giving the improper image header error.

**Reply by fmw42** [reply]
Those must be created differently. You should look into how they are different from your other ones? Do they come from different software from the game? The IM developers will need to look into the header issue to tell you exactly what it thinks is incorrect.  I note that I can open this file on my Mac in GraphicConverter. I suspect that ImageMagick is more strict about conforming to standards and GraphicConverter is not so strict.

**Reply by dlemstra** [reply]
The flags of the file `Apex_Rock_grey_C.dds` are incorrect. The `DDSD_WIDTH (4)` flag is not set and this should be set according to the specification: https://docs.microsoft.com/nl-nl/windows/win32/direct3ddds/dds-header

**Reply by leahanderson1** [reply]
> The flags of the file `Apex_Rock_grey_C.dds` are incorrect. The `DDSD_WIDTH (4)` flag is not set and this should be set according to the specification: https://docs.microsoft.com/nl-nl/windows/win32/direct3ddds/dds-header

lmfao i'm getting this exact same error 5 years later
this is with the game Tomb Raider 1+2+3 Remastered.
